### PR TITLE
Sawa 164 make achievement class

### PR DESCRIPTION
**Pull request by fabiogiorgio**
a little change on where the initAchievement method as ege requested.

### CODE REVIEW

**Comment by egecnr** [comment]
If the user isnt active or if the id is still wrong you are returning the object with http 200 this is sorta dangerous for the frontend since they'd have to do the error handling of the endpoint in the front end for us. Also teachers would criticize you quite a bit since the endpoint design wouldn't be RESTful.

image

Try to return different responses for different cases. Your controller does not handle the errors which may cause a huge problem to you during code review. I see that you're checking if the user is null but you re not taking any precautions if the user is null. If its null then what happens? So what you can do is you can try to improve your endpoint with the checks from the user service. 

You can see here:  Check if the user is active and its an existing user before fetching data.
image

In conclusion, your happy flow is fine but your bad flows are not covered with http responses. You need to enrich your endpoints with http responses that corresponds to correct scenarios. UserController has examples as well as the UserFriendControllers.

**Review by egecnr — CHANGES_REQUESTED** [review]
Check general comments above. If you find a better way to return various http responses feel free to use it and show us too so that we can choose the right formatting for controllers
